Testing/PerformanceTests/analysis.py

```diff
@@ -115,147 +120,96 @@ def get_results_matching(results, field, value):
+def plot_runtime(annotate, saveImage, path, **kwargs):
+    name = kwargs['name']
+    (xData, yData) = get_runtime_data(**kwargs)
+    trace1 = go.Scatter(x=xData, y=yData,
+                        mode='lines+markers',
+                        marker=dict(
+                            size='5',
+                            color="blue"
+                        )
+                        )
+
+    annotations = []
+
+    last_num = kwargs.get('last_num', -1)
+
+    if annotate and not saveImage:
+        # retrieve commitids for annotation on the plotly graph
+        results = get_results(name, type='', where_clause='')
+        sorted = [(res["revision"], res["variables"], res["date"], res) for res in results]
+        sorted.sort(reverse=False)
+
+        commitids = []
+        for (rev, variable, date, res) in sorted:
+            commitids.append("""<a href="%s/commit/%s">  </a>""" % (mantid_addr, res["commitid"]))
+
+        if last_num > 0:
+            commitids = commitids[-last_num:]
+
+        for x, y, text in zip(xData, yData, commitids):
+            annotations.append(
+                dict(
+                    x=x,
+                    y=y,
+                    text=text,
+                    showarrow=False,
+                    font=dict(family='sans serif', size=10),
+                    xanchor='center',
+                    yanchor='center'
+                )
+            )
+
+    title = ''
     if last_num > 0:
-        title("Runtime History of %s (last %d revs)" % (kwargs['name'], kwargs["last_num"]) )
+        title = "Runtime History of %s (last %d revs)" % (name, last_num)
     else:
-        title("Runtime History of %s (all revs)" % kwargs['name'])
-
-
-
-
-
-
-
-
-
-
+        title = "Runtime History of %s (all revs)" % name
+
+    yAxisTitle = 'Runtime/iteration (sec)'
+    xAxisTitle = kwargs['x_field']
+
+    layout = go.Layout(showlegend=False, annotations=annotations,
+                       title=title,
+                       xaxis=dict(title=xAxisTitle),
+                       yaxis=dict(
+                           title=yAxisTitle,
+                           range=[0, np.amax(yData)]
+                       )
+                       )
+    data = [trace1]
+    fig = go.Figure(data=data, layout=layout)
+
+    if saveImage:
+        im_filename = name + ".png"
+        plt.ioff()
+        plt.figure()
+        plt.title(title)
+        plt.xlabel(xAxisTitle)
+        plt.ylabel(yAxisTitle)
+        plt.plot(xData, yData, "-b.")
+        plt.ylim(ymin=0)
+        plt.savefig(os.path.join(path, im_filename))
+        plt.close()
+        return """<img src="%s"/>""" % im_filename
+    else:
+        return offline.plot(fig, output_type='div', show_link=False, auto_open=False, include_plotlyjs=False)
 
 
 # The default HTML header
 default_html_header = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html>
```

Comment:
As these are constants can we move them to the top of the module with the others (and uppercase the variable names). This is the more standard way of using these type of "constants".